src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,183 @@
+package nextstep.subway.line.domain;
+
+import static nextstep.subway.exception.ExceptionMessage.*;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+
+import nextstep.subway.station.domain.Station;
+
+@Embeddable
+public class Sections {
+
+    public static final int SECTION_MIN_COUNT = 1;
+
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    public void add(Section section) {
+        sections.add(section);
+    }
+
+    public List<Station> getStations() {
+        if (sections.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<Station> stations = new ArrayList<>();
+        Section section = extractFirstSection();
+        stations.add(section.getUpStation());
+        Set<Station> upStations = extractUpStations();
+        while (!isLastStation(section.getDownStation(), upStations)) {
+            Section extractSection = extractSectionByContainsUpStation(section.getDownStation());
+            stations.add(extractSection.getUpStation());
+            section = extractSection;
+        }
+        stations.add(section.getDownStation());
+        return stations;
+    }
+
+    public void addSection(Section section) {
+        if (sections.isEmpty()) {
+            sections.add(section);
+            return;
+        }
+        validateAddSection(section);
+        if (!addStationOfBetween(section)) {
+            sections.add(section);
+        }
+    }
+
+    public void removeSection(Station station) {
+        validateRemoveSection(station);
+        Optional<Section> sectionOfUpStation = findSectionOfEqualUpStation(station);
+        Optional<Section> sectionOfDownStation = findSectionOfEqualDownStation(station);
+        if (sectionOfUpStation.isPresent() && sectionOfDownStation.isPresent()) {
+            addConnectSection(sectionOfUpStation.get(), sectionOfDownStation.get());
+        }
+        sectionOfUpStation.ifPresent(it -> sections.remove(it));
+        sectionOfDownStation.ifPresent(it -> sections.remove(it));
+    }
+
+    private boolean addStationOfBetween(Section section) {
+        return addUpStationOfBetween(section) || addDownStationOfBetween(section);
+    }
+
+    private boolean isLastStation(Station downStation, Set<Station> upStations) {
+        return !upStations.contains(downStation);
+    }
+
+    private Section extractSectionByContainsUpStation(Station downStation) {
```

Comment:
```suggestion
    private Section extractSectionByContainsUpStation(Station station) {
```

해당 메서드의 시그니처로 봤을 때 무조건 downStation만 올 수 있는게 아닌 상황에서 downStation과 같은 네이밍을 사용한다면 어떠한 문제가 있을지 고민해보아요

Reply:
실제 메서드는 downStation 이라는 논리적인 부분까지 처리하지는 못하는데, 네이밍으로 인해서 메서드 기능 자체에 대한 오해가 생길듯 합니다.